In [45]:
# Import the necessary dependencies.
import numpy as np
import pandas as pd
import plotly.express as px
import scipy.stats as sts
import plotly.express as plt
import statsmodels.api as sm
import os

In [62]:
#Perform your analysis here
bikecounts_file = os.path.join('../','Resources', 'bikecounts.csv')
bikecounts_df = pd.read_csv(bikecounts_file)
bikecounts_df['Day'] = pd.to_datetime(bikecounts_df['Day'])
bikecounts_df.head()
scatter_fig = plt.scatter(bikecounts_df, y='Brooklyn Bridge', x='High Temp (°F)')
x=bikecounts_df['High Temp (°F)'].astype(float)
scatter_fig.show()
#Does a high temperature affect the number of bike trips across the Brooklyn Bridge?
y= bikecounts_df['Brooklyn Bridge'].astype(float)


In [61]:
model = sm.OLS(y, x)
results=model.fit()
print(results.summary())

                                 OLS Regression Results                                
Dep. Variable:        Brooklyn Bridge   R-squared (uncentered):                   0.914
Model:                            OLS   Adj. R-squared (uncentered):              0.911
Method:                 Least Squares   F-statistic:                              308.6
Date:                Thu, 14 Apr 2022   Prob (F-statistic):                    5.35e-17
Time:                        21:46:31   Log-Likelihood:                         -240.13
No. Observations:                  30   AIC:                                      482.3
Df Residuals:                      29   BIC:                                      483.7
Df Model:                           1                                                  
Covariance Type:            nonrobust                                                  
                     coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------

### Summary
---


Briefly describe your findings here
Hypothesis: If the High temperature increases brooklyn bridge traffic, we would expect to see a growth in brooklyn bridge traffic as the High temperature.

Null Hypothesis: As the High temperature, there will be no effect on brooklyn bridge traffic.

Alternative Hypothesis: As the High temperature increases, brooklyn bridge traffic increases.

Significance Level: 95%

High temperature has a very strong effect relationship with brooklyn bridge traffic such that the adjusted correlation coefficient R^2 is 91.1%; The High temperature variable is also significant for a significance level of 95% since our P value is less than .05.